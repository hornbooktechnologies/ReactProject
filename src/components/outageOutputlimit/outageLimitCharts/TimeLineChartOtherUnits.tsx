import React, { Dispatch, SetStateAction } from 'react'
import ReactApexChart from 'react-apexcharts'
import { TimeLineChartsList } from '../../../models/outageOutputLimit/TimelineChartModel'
import { DateformatConstant, ChartConstant } from '../../../utils/AppConstants'
import { getFiscalYearTimestamp } from '../../../utils/utils'

type Props = {
  className?: string
  timelinecharts: TimeLineChartsList
  selectYear: number  
  setXaxisMin: Dispatch<SetStateAction<number>>
  setXaxisMax: Dispatch<SetStateAction<number>>
  xaxisMin: number
  xaxisMax: number
  noOfUnits: number
}

type xasis = {
  type?: 'category' | 'datetime' | 'numeric' | undefined
}

const TimeLineChartOtherUnits: React.FC<Props> = (props) => {
  const xaxisType: xasis = {
    type: 'datetime',
  }
  const perUnitHeight = 50;
  const marginHeight = 70;
  const yAxisLabelMinWidth = 90;
  const yAxisLabelMaxWidth = 130;
  const subChartHeight: number = marginHeight + (props.noOfUnits * perUnitHeight);
  const state = {
    options: {
      chart: {
        toolbar: {
          tools:{
            download: false,
          },
        },
        zoom: {
          type: 'x',
          autoScaleYaxis: false,
        },
        events: {
          beforeMount: function (chartContext: any, options: any) {
            props.setXaxisMin(options.config.xaxis.min)
            props.setXaxisMax(options.config.xaxis.max)
          },
          beforeZoom: function (chartContext: any, { xaxis }: any) {
            props.setXaxisMin(xaxis.min)
            props.setXaxisMax(xaxis.max)
          },
          beforeResetZoom: function () {
            props.setXaxisMin(getFiscalYearTimestamp(props.selectYear))
            props.setXaxisMax(getFiscalYearTimestamp(props.selectYear + 1))
            return {
              xaxis: {
                min: getFiscalYearTimestamp(props.selectYear),
                max: getFiscalYearTimestamp(props.selectYear + 1),
              },
            }
          },
          scrolled: function (chartContext: any, { xaxis }: any) {
            props.setXaxisMin(xaxis.min)
            props.setXaxisMax(xaxis.max)
          },
        },
      },
      plotOptions: {
        bar: {
          horizontal: true,
          rangeBarGroupRows: true,
        },
      },
      grid: {
        show: true,
        strokeDashArray: ChartConstant.TIMELINE_CHART_GRID_STROKE_AMOUNT,
        borderColor: ChartConstant.TICKLINE_STROKE,
      },
      legend: {
        show: false,
      },
      xaxis: {
        type: xaxisType.type,
        min: props.xaxisMin,
        max: props.xaxisMax,
        labels: {
          datetimeFormatter: {
            year: DateformatConstant.M_D,
            month: DateformatConstant.M_D,
            day: DateformatConstant.M_D,
            hour: DateformatConstant.HH_MM,
          },
          style: {
            colors: ChartConstant.WHITE_STROKE,
          },
        },
        axisBorder: {
          show: false,
        },
      },
      yaxis: {
        showForNullSeries: false,
        labels: {
          minWidth: yAxisLabelMinWidth,
          maxWidth: yAxisLabelMaxWidth,
          style: {
            colors: ChartConstant.WHITE_STROKE,
          },
          offsetX: -5,
        },
      },
      tooltip: {
        x: {
          format: DateformatConstant.TIMELINE_CHART_TOOLTIP_FORMAT,
        },
      },
      fill: {
        opacity: ChartConstant.OPACITY,
      },
    },
    series: props.timelinecharts,
  }

  if (props.timelinecharts.length > 0) {
    return (
      <ReactApexChart
        className={['timeline-chart', props.className].join(' ')}
        options={state.options}
        series={state.series}
        type="rangeBar"
        height={subChartHeight}
      />
    )
  } else {
    return null
  }
}

export default TimeLineChartOtherUnits
