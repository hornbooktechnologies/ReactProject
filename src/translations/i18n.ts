import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
