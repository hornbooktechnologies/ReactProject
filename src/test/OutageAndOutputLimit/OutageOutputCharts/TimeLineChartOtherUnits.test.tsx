import React from "react";
import renderer from "react-test-renderer";
import TimeLineChartOtherUnits from "../../../components/outageOutputlimit/outageLimitCharts/TimeLineChartOtherUnits";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: TimeLineChartOtherUnits", () => {
  it("TimeLineChartOtherUnits renders correctly", () => {
    const chartTestData = {
        timelinecharts: [],
        selectYear: 2021,
        xaxisMin: 2021,
        xaxisMax: 2022,
        setXaxisMin: () => {},
        setXaxisMax: () => {},
        noOfUnits: 4
    }
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <TimeLineChartOtherUnits {...chartTestData} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
