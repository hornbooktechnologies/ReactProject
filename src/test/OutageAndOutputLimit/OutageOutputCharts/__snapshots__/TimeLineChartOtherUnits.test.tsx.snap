// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TimeLineChartOtherUnits TimeLineChartOtherUnits renders correctly 1`] = `null`;
